packages/context-editor/src/plugins/code/index.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * Based on https://gitlab.com/emergence-engineering/prosemirror-codemirror-block/-/blob/main/src/index.ts
+ *
+ * Differences:
+ * * Adds createSelect
+ * * Exports default with options
+ * * No theme
+ * */
+
+import { redo, undo } from "prosemirror-history";
+import { Node, Schema } from "prosemirror-model";
+import { Plugin, PluginKey } from "prosemirror-state";
+import { EditorView } from "prosemirror-view";
+
+import type { CodeBlockSettings, LanguageLoaders } from "./types";
+import { codeMirrorBlockNodeView } from "./codeMirrorBlockNodeView";
+import { defaultSettings } from "./defaults";
+import languageLoaders from "./languageLoaders";
+import { CodeBlockLanguages } from "./languages";
+import parsers from "./parsers";
+import { codeBlockArrowHandlers } from "./utils";
+
+export const codeMirrorBlockKey = new PluginKey("codemirror-block");
+
+const codeMirrorBlockPlugin = (settings: CodeBlockSettings) => {
+	return new Plugin({
+		key: codeMirrorBlockKey,
+		props: {
+			nodeViews: {
+				// @ts-expect-error Type FIXME: '(pmNode: Node, view: PMEditorView, getPos: (() => number) | boolean) => NodeView' is not assignable to type 'NodeViewConstructor'.
+				code_block: codeMirrorBlockNodeView(settings),
+			},
+		},
+	});
+};
+
+export {
+	codeMirrorBlockNodeView,
+	codeBlockArrowHandlers,
+	codeMirrorBlockPlugin,
+	type CodeBlockSettings,
+	type LanguageLoaders,
+	CodeBlockLanguages,
+	defaultSettings,
+	parsers,
```

Comment:
as per this comment (https://github.com/pubpub/platform/pull/1013#discussion_r1981153261), the dynamic import gains can be made by just removing the parsers export

```suggestion
import { codeBlockArrowHandlers } from "./utils";

export const codeMirrorBlockKey = new PluginKey("codemirror-block");

const codeMirrorBlockPlugin = (settings: CodeBlockSettings) => {
	return new Plugin({
		key: codeMirrorBlockKey,
		props: {
			nodeViews: {
				// @ts-expect-error Type FIXME: '(pmNode: Node, view: PMEditorView, getPos: (() => number) | boolean) => NodeView' is not assignable to type 'NodeViewConstructor'.
				code_block: codeMirrorBlockNodeView(settings),
			},
		},
	});
};

export {
	codeMirrorBlockNodeView,
	codeBlockArrowHandlers,
	codeMirrorBlockPlugin,
	type CodeBlockSettings,
	type LanguageLoaders,
	CodeBlockLanguages,
	defaultSettings,
```